### PR TITLE
Fix istioctl analyze having new config store errors

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Fix https://github.com/istio/istio/issues/36160

This fix is consistent with the old `ApplyContent` logic.

/cc @therealmitchconnors

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
CAn you add a test?

**Comment by monsteredc** [comment]
@howardjohn PTAL. Thanks

**Comment by monsteredc** [comment]
Revised with the logic: if the config exists, set the `ResourceVersion` of the new config to the old one in order to perform successful update.

**Comment by hanxiaop** [comment]
not stale

**Comment by hanxiaop** [comment]
not stale

**Comment by hanxiaop** [comment]
not stale

**Comment by hanxiaop** [comment]
https://github.com/istio/istio/blob/d2b96bb9572875fee561c7c34e1e42f58e04c88c/pilot/pkg/config/memory/store.go#L204-L212
Looked through the code, the error is due to the conflict in update. Set the newer config `ResourceVersion` to the annotation part can avoid of `ResourceVersion` conflicts.

File: pilot/pkg/config/file/store.go [hunk 1]
```diff
@@ -204,7 +204,14 @@ func (s *KubeSource) ApplyContent(name, yamlText string) error {
...
 		if !found || oldSha != r.sha {
 			s.versionCtr++
-			r.config.ResourceVersion = fmt.Sprintf("v%d", s.versionCtr)
+			if !found {
+				r.config.ResourceVersion = fmt.Sprintf("v%d", s.versionCtr)
+			} else {
+				if r.config.Annotations == nil {
+					r.config.Annotations = map[string]string{}
+				}
+				r.config.Annotations[memory.ResourceVersion] = fmt.Sprintf("v%d", s.versionCtr)
```
Comment (howardjohn) [inline pilot/pkg/config/file/store.go:213]: I get why we need L208, but not why we need this. Can you add test/comments

**Comment by hanxiaop** [comment]
not stale

**Comment by zirain** [comment]
not stale

File: pilot/pkg/config/file/store.go [hunk 2]
```diff
@@ -204,7 +204,17 @@ func (s *KubeSource) ApplyContent(name, yamlText string) error {
 		oldSha, found := s.shas[key]
 		if !found || oldSha != r.sha {
 			s.versionCtr++
-			r.config.ResourceVersion = fmt.Sprintf("v%d", s.versionCtr)
+			if !found {
+				r.config.ResourceVersion = fmt.Sprintf("v%d", s.versionCtr)
+			} else {
+				// If the old resource version exists, and we want to overwrite the old object to the new object,
+				// we should set the resource version in the annotation to avoid of conflicting resource reported
```
Comment (therealmitchconnors) [inline pilot/pkg/config/file/store.go:211]: If I understand this correctly, we are allowing subsequent copies of a resource to override the initial copies.  Is that correct?  If so, what determines the order?

Comment (hanxiaop) [reply pilot/pkg/config/file/store.go:211]: @therealmitchconnors Yes correct. The newest content will be the one to be analyzed. The logic I refered to was from the old branch before this https://github.com/istio/istio/pull/35764 got merged. Which explained here:

https://github.com/istio/istio/blob/842115039209d938d6bc9f96736f1ca62e2f8f6f/galley/pkg/config/source/kube/inmemory/kubesource.go#L142-L146

And the code is:

https://github.com/istio/istio/blob/842115039209d938d6bc9f96736f1ca62e2f8f6f/galley/pkg/config/source/kube/inmemory/kubesource.go#L160-L166

**Comment by hanxiaop** [comment]
not stale

**Pull request closed without merging**